components/scream/src/share/util/scream_utils.hpp

```diff
@@ -75,6 +75,28 @@ int strcmp(const char* first, const char* second)
   }
   return *(const unsigned char*)first - *(const unsigned char*)second;
 }
+template<class T>
+KOKKOS_INLINE_FUNCTION
+const T* upper_bound(const T* first, const T* last, const T& value)
```

Comment:
It would be good to make a unit test for upper_bound.

Reply:
Done.